src/test/java/core/basesyntax/strategy/transactionhandlers/PurchaseTransactionHandlerImplTest.java

```diff
@@ -0,0 +1,74 @@
+package core.basesyntax.strategy.transactionhandlers;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.dao.FruitStorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class PurchaseTransactionHandlerImplTest {
+    private static final int FRUIT_QUANTITY = 50;
+    private static final String FRUIT_NAME = "banana";
+    private FruitTransaction fruit;
+    private FruitStorageDao dao;
+    private TransactionHandler purchase;
+
+    @Before
+    public void setUp() throws Exception {
+        Storage.storage.put(FRUIT_NAME, FRUIT_QUANTITY);
+        fruit = new FruitTransaction(FruitTransaction.Operation.PURCHASE,
+                FRUIT_NAME,10);
+        dao = new FruitStorageDaoImpl();
+        purchase = new PurchaseTransactionHandlerImpl();
+    }
+
+    @Test
+    public void addFruitName_Ok() {
+        purchase.transaction(fruit);
+        List<String> expected = List.of(fruit.getName());
+        List<String> actual = dao.getAllFruitsNames();
+        Assert.assertEquals(expected,actual);
+    }
+
+    @Test(expected = NullPointerException.class)
```

Comment:
```suggestion
    @Test(expected = RuntimeException.class)
```
you should never expect your program to throw NPE, throw a RuntimeException rather than NPE